gemfirexd/tools/src/dunit/java/com/pivotal/gemfirexd/transactions/MVCCDUnit.java

```diff
@@ -51,6 +51,17 @@ protected String reduceLogging() {
     return "fine";
   }
 
+  public static void validateNoActiveSnapshotTx() {
+    TXManagerImpl txMgr = Misc.getGemFireCache().getCacheTransactionManager();
+    if (txMgr != null) {
+      Iterator<TXStateProxy> itr = txMgr.getHostedTransactionsInProgress().iterator();
+      while (itr.hasNext()) {
+        TXStateProxy tx = itr.next();
+        if (tx.isSnapshot())
+          assertTrue("tx is not closed.", tx.isClosed());
```

Comment:
Change is fine but just a doubt. Is there a possibility that a closed txstateproxy is caught in the iterator. Can't we just check the size of hostedTxnInProgress.